components/scream/src/share/field/field_boundedness_check.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef SCREAM_FIELD_BOUNDEDNESS_CHECK_HPP
+#define SCREAM_FIELD_BOUNDEDNESS_CHECK_HPP
+
+#include "share/field/field.hpp"
+#include "ekat/util/ekat_math_utils.hpp"
+
+namespace scream
+{
+
+// This field property check returns true if all data in a given field are
+// bounded by the given upper and lower bounds (inclusively), and false if not.
+// It can repair a field that fails the check by clipping all unbounded values
+// to their closest bound.
+template<typename ScalarType, typename Device>
+class FieldBoundednessCheck: public FieldPropertyCheck<ScalarType, Device> {
+public:
+
+  // No default constructor -- we need lower and upper bounds.
+  FieldBoundednessCheck () = delete;
+
+  // Constructor with lower and upper bounds -- can repair fields that fail the check
+  // by overwriting nonpositive values with the given lower bound.
+  explicit FieldBoundednessCheck (ScalarType lower_bound,
+                                  ScalarType upper_bound) :
+    m_lower_bound(lower_bound),
+    m_upper_bound(upper_bound) {
+    EKAT_ASSERT_MSG(lower_bound < upper_bound,
```

Comment:
I can foresee one trying to force a field to be constant, without changing any of the rest of the code. Perhaps when trying to see the impact of an input field, or to debug something. He/she might use a FPC with can_repair=true to enforce this.

If we think that might be a legit use of FPC, we should allow lb==ub.

Reply:
I agree.